### PR TITLE
trace: upgrade Sentry client to have better reporting

### PR DESCRIPTION
**Pull request by unknwon**
This is the first step of combining the following two efforts ([full plan](https://docs.google.com/document/d/102xx77VWab6a50EEnFKsgRO2FdIhh1YNiqmhnVFW4FA/edit#)):

1. Start using https://github.com/cockroachdb/errors (see rationale below)
1. Replace our legacy and unmaintained Sentry client ("github.com/getsentry/raven-go") to "github.com/cockroachdb/sentry-go" that comes with "github.com/cockroachdb/errors" (a fork that is not the latest but still much better than what we have).

### Rationale for 1.

After reading "github.com/cockroachdb/errors"'s [RFC](https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20190318_error_handling.md) and its README, it looks very promising and close to what I would imagine we should be doing for errors going forward. 

In particular (there are much more provided):

- Native Sentry support
- PII-aware: strip when report to Sentry, print full in logs
- Extensibility
- Drop-in replacement for "github.com/pkg/errors" and compatible with _both_ "github.com/pkg/errors" and Go 1.13 errors interface.
- Preserve error equivalences across network boundaries
- Error domain (example use case: prevent expose errors from dependent services directly to clients, e.g. repo-updater returns gitserver error to frontend, controversially, the error should be handled within repo-updater)

---

**Before (no context line):**

CleanShot 2020-12-18 at 18 06 15@2x

**After:**

CleanShot 2020-12-18 at 18 02 34@2x

_Part of #16109_

### CODE REVIEW

File: internal/trace/httptrace.go [hunk 1]
```diff
@@ -227,11 +198,11 @@ func HTTPTraceMiddleware(next http.Handler) http.Handler {
 		log15.Debug("TRACE HTTP",
 			"method", r.Method,
 			"url", r.URL.String(),
-			"routename", routeName,
+			"route_name", routeName,
```
Comment (unknwon) [inline internal/trace/httptrace.go:201]: snake case tags look more conventional in Sentry.

**Review by keegancsmith — COMMENTED** [review]
Haven't reviewed code (might not get around to it before vacation). But read description. Sounds good moving to using cockroachdb/errors pkg. I looked at it a while back and thought it was great. :+1:

File: internal/sentry/sentry.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+			if conf.Get().Log == nil {
+				return
+			}
+
+			if conf.Get().Log.Sentry == nil {
+				return
+			}
+
+			// An empty dsn value is ignored: not an error.
+			if err := initClient(conf.Get().Log.Sentry.Dsn); err != nil {
```
Comment (tsenart) [inline internal/sentry/sentry.go:68]: Is this safe to be called from a go-routine?

**Review by tsenart — APPROVED** [review]

File: internal/sentry/sentry.go (hunk 2 above)
Comment (unknwon) [reply internal/sentry/sentry.go:68]: I believe so, the Sentry client internally uses mutex: https://sourcegraph.com/github.com/cockroachdb/sentry-go@master/-/blob/hub.go#L59-62

**Pull request merged**